lib/mix/tasks/config_seed.ex

```diff
@@ -0,0 +1,72 @@
+defmodule Mix.Tasks.ConfigSeed do
+  @moduledoc """
+  Add services from configuration.
+  """
+
+  use Mix.Task
+
+  import Ecto.Query
+
+  alias Uptime.Repo
+
+  require Logger
+
+  @requirements ["app.config"]
+
+  @impl Mix.Task
+  def run(_args) do
+    load()
+    existing_ids = Repo.all(from(s in Uptime.Service, select: s.id))
+
+    :services
+    |> Uptime.Env.get()
+    |> Enum.map(&save_service(&1, existing_ids))
+    |> Enum.filter(fn {result, _} -> result == :error end)
+    |> Enum.map(fn {_result, changeset} -> changeset_error_to_string(changeset) end)
+    |> Enum.each(&Logger.error/1)
+  end
+
+  defp load do
+    Application.load(:uptime)
+    Application.ensure_all_started([:postgrex, :ecto])
+    Repo.start_link(pool_size: 2)
+  end
+
+  defp save_service(attrs, existing_ids) do
+    attrs
+    |> Uptime.Service.changeset()
+    |> Repo.insert(
+      on_conflict: {:replace_all_except, [:id, :inserted_at]},
+      conflict_target: :slug,
+      returning: true
+    )
+    |> tap(fn
+      {:ok, service} ->
+        unless Enum.member?(existing_ids, service.id) do
+          create_job(service.id)
+        end
```

Comment:
is this necessary or does oban already check for this because the job is marked as unique?

Reply:
oban handles the unique for us and even returns on ok tuple (with a field set that you hit the unique)

Reply:
This is all I'm seeing and it creates an additional job

```
Oban.insert() #=> {:ok,
 %Oban.Job{
   __meta__: #Ecto.Schema.Metadata<:loaded, "public", "oban_jobs">,
   id: 1384,
   state: "scheduled",
   queue: "default",
   worker: "Uptime.CheckJob",
   args: %{"id" => "svc_01J7487V7BWAWC91000M4D9HAN"},
   meta: %{},
   tags: [],
   errors: [],
   attempt: 0,
   attempted_by: nil,
   max_attempts: 20,
   priority: 0,
   attempted_at: nil,
   cancelled_at: nil,
   completed_at: nil,
   discarded_at: nil,
   inserted_at: ~U[2024-09-06 20:01:29.827509Z],
   scheduled_at: ~U[2024-09-06 20:01:30.770909Z],
   conf: nil,
   conflict?: true,
   replace: nil,
   unique: nil,
   unsaved_error: nil
 }}
```

Reply:
Maybe because it's `conflict?: true` that it sorts it out at runtime even though an additional job is created.